File: November_Leetcode_Challenge_2022/Day4/345._Reverse_Vowels_of_a_String.java
class Solution {
    public String reverseVowels(String s) {
        char []c = s.toCharArray();
        int start=0;
        int end=s.length()-1;
        while(start<end){
            while(start<end && c[start]!='a' && c[start]!='e' && c[start]!='i' && c[start]!='o' && c[start]!='u' && c[start]!='A' && c[start]!='E' && c[start]!='I' && c[start]!='O' && c[start]!='U')start++;
 
            while(start<end && c[end]!='a' && c[end]!='e' && c[end]!='i' && c[end]!='o' && c[end]!='u' && c[end]!='A'&& c[end]!='E' && c[end]!='I' && c[end]!='O' && c[end]!='U')end--;
            if(start<end){
                char temp=c[start];
                c[start]=c[end];
                c[end]=temp;
                start++;
                end--;
            }
 
        }
        return new String(c);
    }
}

File: November_Leetcode_Challenge_2022/Day4/345._Reverse_Vowels_of_a_String.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c == 'a' || c == 'i' || c == 'e' || c == 'o' || c == 'u'
            || c == 'A' || c == 'I' || c == 'E' || c == 'O' || c == 'U';
    }
    string reverseVowels(string s) {
      int i = 0,j=s.size();
      while(i<j)
      {
          if(isVowel(s[i]) && isVowel(s[j]))
          {
              swap(s[i],s[j]);
              i++;
              j--;
          }
          else if(!isVowel(s[i]))
          {
            i++;  
          }
          else if(!isVowel(s[j]))
          {
              j--;
          }
      }
        return s;
    }
};
